lib/puppet/file_serving/http_metadata.rb

```diff
@@ -15,13 +15,25 @@ def initialize(http_response, path = '/dev/null')
     # use a default mtime in case there is no usable HTTP header
     @checksums[:mtime] = "{mtime}#{Time.now}"
 
+    # RFC-1864, deprecated in HTTP/1.1 due to partial responses
     checksum = http_response['content-md5']
     if checksum
       # convert base64 digest to hex
       checksum = checksum.unpack("m").first.unpack("H*").first
       @checksums[:md5] = "{md5}#{checksum}"
     end
 
+    {
+      md5: 'X-Checksum-Md5',
+      sha1: 'X-Checksum-Sha1',
+      sha256: 'X-Checksum-Sha256'
+    }.each_pair do |checksum_type, header|
+      checksum = http_response[header]
+      if checksum
+        @checksums[checksum_type] = "{#{checksum_type}}#{checksum}"
```

Comment:
@MikaelSmith there was some discussion in the original PR about making the `X-Checksum-*` headers configurable, thoughts?

Reply:
Looking at my previous comments, `X-Checksum-*` didn't seem to be a common standard. Maybe I was mistaken or that's changed. My main consideration was to be able to customize it if that's the case so that we don't have incorrect behavior from e.g. `X-Checksum-Sha256` if it has an unexpected format. I guess I'd be surprised if that actually comes up though.

Reply:
At this stage we're just extracting the checksum value, but not trying to parse it. It's not until the next call to `collect` that we try to determine which of the checksum types to use. If one of the `X-Checksum-*` headers was incorrect, the user could specify a preferred `checksum` type, and worst case specify `mtime` if there are not suitable hash checksums.